single.php

```diff
@@ -130,6 +130,35 @@
 
 if ( post_password_required( $post->ID ) ) {
 
+	// Check if page url has a unique id(custom hash), appended with it, if not add one.
+	$custom_hash = filter_input( INPUT_GET, 'ch', FILTER_SANITIZE_STRING );
+	if ( ! $custom_hash ) {
+		wp_safe_redirect( add_query_arg( 'ch', md5( uniqid( '', true ) ), get_permalink() ) );
+		exit();
+	}
+
+	/**
+	 * Password protected form validation:
+	 * The latest entered password is stored as a secure hash in a cookie named 'wp-postpass_' . COOKIEHASH.
+	 * When the password form is called, that cookie has been validated already by WordPress.
+	 */
+	$old_cookie = get_transient( 'p4-postpass_' . $custom_hash );
+	if ( false === $old_cookie ) {
+		// this code runs when there is no valid transient set.
+		if ( isset( $_COOKIE[ 'wp-postpass_' . COOKIEHASH ] ) ) {
```

Comment:
If we put `if ( isset( $_COOKIE[ 'wp-postpass_' . COOKIEHASH ] ) ) {` on the outside, then we only need to have it once.

```php
if ( isset( $_COOKIE[ 'wp-postpass_' . COOKIEHASH ] ) ) {
    $current_cookie = wp_unslash( $_COOKIE[ 'wp-postpass_' . COOKIEHASH ] );
    set_transient( 'p4-postpass_' . $custom_hash, $current_cookie, $expiration = 60 * 5 );
    if ( false === $old_cookie ) {...
    } else {...
    }
}